guide/phone_store/phone_store_service.bal

```diff
@@ -61,171 +52,177 @@ import ballerina/log;
+
+// Service endpoint.
+listener http:Listener httpListener = new(9090);
+
+http:Request backendreq = new;
 
-// phone store service, which allows users to order phones online for delivery
-@http:ServiceConfig { basePath: "/phonestore" }
-service<http:Service> phoneStoreService bind listener {
-    // Resource that allows users to place an order for a phone
+// Phone store service, which allows users to order phones online for delivery.
+@http:ServiceConfig {
+    basePath: "/phonestore"
+}
+service phoneStoreService on httpListener {
+    // Resource that allows users to place an order for a phone.
     @http:ResourceConfig {
         methods: ["POST"],
         consumes: ["application/json"],
         produces: ["application/json"]
     }
-    placeOrder(endpoint caller, http:Request request) {
-
+    resource function placeOrder(http:Caller caller, http:Request request) {
         backendreq = untaint request;
-        http:Response response;
-        PhoneOrder newOrder;
-        json reqPayload;
-
-        // Try parsing the JSON payload from the request
-        match request.getJsonPayload() {
-            // Valid JSON payload
-            json payload => reqPayload = payload;
-            // NOT a valid JSON payload
-            any => {
-                response.statusCode = 400;
-                response.setJsonPayload({ "Message": "Invalid payload - Not a valid JSON payload" });
-                _ = caller->respond(response);
-                done;
-            }
+        http:Response response = new;
+        PhoneOrder newOrder = {};
+        json requestPayload = {};
+
+        var payload = request.getJsonPayload();
+        // Try parsing the JSON payload from the request.
+        if (payload is json) {
+            // Valid JSON payload.
+            requestPayload = payload;
+        } else {
+            // NOT a valid JSON payload.
+            response.statusCode = 400;
+            response.setJsonPayload({ "Message": "Invalid payload - Not a valid JSON payload" });
+            checkpanic caller->respond(response);
         }
-        json name = reqPayload.Name;
-        json address = reqPayload.Address;
-        json contact = reqPayload.ContactNumber;
-        json phoneName = reqPayload.PhoneName;
 
-        // If payload parsing fails, send a "Bad Request" message as the response
+        json name = requestPayload.Name;
+        json address = requestPayload.Address;
+        json contact = requestPayload.ContactNumber;
+        json phoneName = requestPayload.PhoneName;
+
+        // If payload parsing fails, send a "Bad Request" message as the response.
         if (name == null || address == null || contact == null || phoneName == null) {
             response.statusCode = 400;
             response.setJsonPayload({ "Message": "Bad Request - Invalid payload" });
-            _ = caller->respond(response);
-            done;
+            checkpanic caller->respond(response);
```

Comment:
I think previously `done` was there for a reason. Can we verify please?

Reply:
Added `return` clause